### PR TITLE
feat: resolve token symbols (SOL, USDC, ETH, etc.) to addresses

### LINKED ISSUE #88: UX: Symbol shortcuts (SOL, USDC, ETH) should resolve to token addresses
## Description

Currently `--from SOL --to USDC` fails with:

```
Error: fromTokenAddress must be a valid Solana address
```

Users must provide full contract addresses like `So11111111111111111111111111111111111111112` and `EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v`.

## Expected Behavior

The CLI should resolve common symbols (SOL, ETH, USDC, USDT, etc.) to their canonical token addresses per chain. For example:

- `--chain solana --from SOL` → `So11111111111111111111111111111111111111112`
- `--chain base --from ETH` → `0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee`
- `--chain solana --to USDC` → `EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v`

This is especially important for AI agent usage where symbol names are more natural than raw addresses.

## Suggestion

Add a symbol-to-address lookup table for top tokens per chain. Fall through to raw address if no match.

Found during QA testing of v1.8.0.

### PR DESCRIPTION
**Pull request by Codier**
## Summary

Adds symbol shortcuts so users can write `--from SOL --to USDC` instead of raw contract addresses.

### Changes

- **`TOKEN_SYMBOLS` lookup table** — maps common symbols (SOL, WSOL, ETH, WETH, BNB, WBNB, USDC, USDT) to canonical addresses per chain (solana, ethereum, base, bsc)
- **`resolveTokenAddress(symbolOrAddress, chain)`** — resolves symbols to addresses, falls through to raw address if no match
- **Wired into quote command** — `--from` and `--to` are resolved before hitting the API
- **Updated help text** with symbol examples
- **Tests added** for resolveTokenAddress (symbol resolution, case-insensitivity, passthrough, edge cases)

### Examples

```bash
# Before
nansen quote --chain solana --from So11111111111111111111111111111111111111112 --to EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v --amount 1000000000

# After
nansen quote --chain solana --from SOL --to USDC --amount 1000000000
```

All 622 tests passing ✅

Closes #88

### CODE REVIEW

File: src/trading.js [hunk 1]
```diff
@@ -30,6 +30,49 @@ const WRAPPED_NATIVE_TOKENS = {
   bsc:      { address: '0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c', symbol: 'WBNB', nativeSymbol: 'BNB' },
 };
 
+// Common token symbol → address lookup per chain.
+// Native sentinels: Solana uses native mint, EVM uses 0xeee…eee.
+const EVM_NATIVE = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee';
+const TOKEN_SYMBOLS = {
+  solana: {
+    SOL:  'So11111111111111111111111111111111111111112',
+    WSOL: 'So11111111111111111111111111111111111111112',
```
Comment (Codier) [inline src/trading.js:41]: 🟡 **Should Fix — DRY**: `TOKEN_SYMBOLS` duplicates addresses already in `WRAPPED_NATIVE_TOKENS` (WETH on base/ethereum, WBNB on bsc). If these ever diverge it'll be a subtle bug. Consider deriving the wrapped entries from `WRAPPED_NATIVE_TOKENS` or at least adding a comment noting the duplication.

File: src/trading.js [hunk 2]
```diff
@@ -30,6 +30,49 @@ const WRAPPED_NATIVE_TOKENS = {
...
+    USDC: 'EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v',
+    USDT: 'Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB',
+  },
+  ethereum: {
+    ETH:  EVM_NATIVE,
+    WETH: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
+    USDC: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
+    USDT: '0xdac17f958d2ee523a2206206994597c13d831ec7',
```
Comment (Codier) [inline src/trading.js:49]: 🔴 **Blocking — Correctness**: Base USDT address `0xfde4C96c8593536E31F229EA8f37b2ADa2699bb2` — is this the canonical/bridged USDT on Base? The common one is the Stargate bridged USDT. Worth double-checking, because a wrong address here silently routes user funds to the wrong token. (Ethereum/Solana USDC/USDT addresses look correct.)

File: src/trading.js [hunk 3]
```diff
@@ -30,6 +30,49 @@ const WRAPPED_NATIVE_TOKENS = {
...
+  },
+  base: {
+    ETH:  EVM_NATIVE,
+    WETH: '0x4200000000000000000000000000000000000006',
+    USDC: '0x833589fcd6edb6e08f4c7c32d4f71b54bda02913',
+    USDT: '0xfde4C96c8593536E31F229EA8f37b2ADa2699bb2',
```
Comment (Codier) [inline src/trading.js:57]: 🟢 **Nitpick**: Confirmed that `execute` uses quote IDs not token addresses, so only wiring `quote` is correct. But the help text for `execute` still shows raw addresses in its examples — might be nice to update for consistency, though not blocking.

File: src/__tests__/trading.test.js [hunk 4]
```diff
@@ -26,6 +26,7 @@ import {
   stripLeadingZeros,
   buildTradingCommands,
   getWrappedNativeFromWarning,
+  resolveTokenAddress,
 } from '../trading.js';
```
Comment (Codier) [inline src/__tests__/trading.test.js:30]: 🟢 **Praise**: Clean, concise test suite — good coverage of happy path, case-insensitivity, passthrough, and null handling without bloat. 👍

File: src/trading.js (hunk 2 above)
Comment (Codier) [inline src/trading.js:49]: ✅ Verified — `0xfde4C96c8593536E31F229EA8f37b2ADa2699bb2` is the [bridged USDT on Base per BaseScan](https://basescan.org/token/0xfde4C96c8593536E31F229EA8f37b2ADa2699bb2). My concern was unfounded. Downgrading from 🔴 to ✅.

One thing to note: this is the **legacy L2 bridge** USDT, not a native Tether deployment. If Tether deploys natively on Base in the future (like USDC did), this address will need updating. Consider adding a comment in the code noting this.
